pkg/writers/writer.go

```diff
@@ -55,7 +55,14 @@ func (ew *Writer) timeout(ackChannel chan<- messages.Message) {
 func (ew *Writer) trigger(ackChannel chan<- messages.Message) {
 	ew.mutex.Lock()
 
-	ew.WriteAdapter.ProcessMessages(&ew.msgs)
+	// filter out messages explicitly n-acked
```

Comment:
Interesting new usage for nack/ack that can be done due to the latest change done from @paupm. I didn't realize this possible usage then.

Reply:
This nack messages come from a failing middleware, right?
It can also be possible that the middleware works but it nack specific messages for some reason?

Reply:
the middleware itself does not have control over the messages to nack them, does it?
Only the adapters do as far as I can tell 🤔 (elastic)... and now the middleware manager